cardinal/types/component/utils.go

```diff
@@ -0,0 +1,10 @@
+package component
+
+// CastComponentMetadatasToComponents Cast an array of ComponentMetadata into an array of Component
+func CastComponentMetadatasToComponents(comps []ComponentMetadata) []Component {
+	ret := make([]Component, len(comps))
+	for i, comp := range comps {
+		ret[i] = comp
+	}
+	return ret
+}
```

Comment:
smol nit: we use the word `Convert` for functions like this - and this this isn't necessarily casting anything in the pure sense of the word, i think convert would be better fitting